vlm/src/main/scala/geotrellis/contrib/vlm/avro/GeoTrellisDataPath.scala

```diff
@@ -0,0 +1,71 @@
+package geotrellis.contrib.vlm.avro
+
+import geotrellis.contrib.vlm.DataPath
+
+
+/** Represents a path that points to a GeoTrellis layer saved in a catalog.
+ *
+ *  @param path Path to the layer. This can be either an Avro or COG layer.
+ *    The given path needs to be in a `URI` format that include the following query
+ *    parameters:
+ *      - '''layer''': The name of the layer.
+ *      - '''zoom''': The zoom level to be read.
+ *      - '''band_count''': The number of bands of each Tile in the layer.
+ *    Of the above three parameters, `layer` and `zoom` are required. In addition,
+ *    this path can be prefixed with, '''gt+''' to signify that the target path
+ *    is to be read in only by [[GeotrellisRasterSource]].
+ *  @example "s3://bucket/catalog?layer=layer_name&zoom=10"
+ *  @example "hdfs://data-folder/catalog?layer=name&zoom-12&band_count=5"
+ *  @example "gt+file:///tmp/catalog?layer=name&zoom=5"
+ *  @note The order of the query parameters does not matter.
+ */
+case class GeoTrellisDataPath(val path: String) extends DataPath {
```

Comment:
I want to point out here that the files/types that I've added concerning GT are spelled differently than other files in the same package. Mine are spelled as: `GeoTrellis` whereas the other files are: `Geotrellis`. I don't feel too strongly about either or, but `GeoTrellis` is the correct spelling, so I wasn't sure if that's something we'd want to change to.

Reply:
`GeoTrellis` is a good thing, let's standardize on it